### PR TITLE
Move functional tests to BrowserStack

### PR DESCRIPTION
**Pull request by georgiy-abbasov**

### CODE REVIEW

**Comment by georgiy-abbasov** [comment]
@testcafe-build-bot /retest

**Comment by georgiy-abbasov** [comment]
@testcafe-build-bot \retest

**Comment by georgiy-abbasov** [comment]
@testcafe-build-bot \retest

**Comment by georgiy-abbasov** [comment]
@testcafe-build-bot \retest

**Comment by georgiy-abbasov** [comment]
@testcafe-build-bot \retest

**Comment by georgiy-abbasov** [comment]
@testcafe-build-bot \retest

**Comment by georgiy-abbasov** [comment]
@testcafe-build-bot \retest

File: Gulpfile.js [hunk 1]
```diff
@@ -543,8 +543,9 @@ gulp.task('publish-website', ['build-website-production'], function () {
 gulp.task('test-docs-travis', ['test-website-travis', 'lint']);
 
 
-function testFunctional (fixturesDir, testingEnvironmentName) {
+function testFunctional (fixturesDir, testingEnvironmentName, browserProviderName) {
     process.env.TESTING_ENVIRONMENT = testingEnvironmentName;
+    process.env.browserProvider     = browserProviderName;
```
Comment (AlexanderMoskovkin) [inline Gulpfile.js]: Use UPPER_CASE for the env variable

File: package.json [hunk 2]
```diff
@@ -108,6 +108,7 @@
     "basic-auth": "^1.1.0",
     "body-parser": "^1.17.1",
     "broken-link-checker": "^0.7.0",
+    "browserstack-connector": "0.1.0",
```
Comment (AlexanderMoskovkin) [inline package.json]: use `^` 
It helps to have fixes automatically

File: test/functional/config.js [hunk 3]
```diff
@@ -4,80 +4,90 @@ var isTravisEnvironment = !!process.env.TRAVIS;
...
+    browserstack: 'browserstack'
+};
+
 var testingEnvironmentNames = {
-    saucelabsOSXDesktopAndMSEdgeBrowsers: 'saucelabs-osx-desktop-and-ms-edge-browsers',
-    saucelabsMobileBrowsers:              'saucelabs-mobile-browsers',
-    localBrowsers:                        'local-browsers',
-    oldBrowsers:                          'old-browsers',
-    legacy:                               'legacy'
+    OSXDesktopAndMSEdgeBrowsers: 'osx-desktop-and-ms-edge-browsers',
```
Comment (AlexanderMoskovkin) [inline test/functional/config.js]: `osXDesktopAndMSEdgeBrowsers`

**Comment by georgiy-abbasov** [comment]
@testcafe-build-bot \retest

**Comment by georgiy-abbasov** [comment]
@testcafe-build-bot \retest

**Pull request merged**